commands/src/main/java/org/wildfly/extras/creaper/commands/deployments/Deploy.java

```diff
@@ -0,0 +1,206 @@
+package org.wildfly.extras.creaper.commands.deployments;
+
+import org.jboss.as.controller.client.OperationBuilder;
+import org.jboss.dmr.ModelNode;
+import org.wildfly.extras.creaper.core.online.Constants;
+import org.wildfly.extras.creaper.core.online.OnlineCommand;
+import org.wildfly.extras.creaper.core.online.OnlineCommandContext;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.ENABLED;
+
+/**
+ * Command which takes care about deploying specified deployment under specified deployment name.
+ * <p/>
+ * In case of domain it is deployed by default to all server groups
+ */
+public final class Deploy implements OnlineCommand {
+
+    private final String deploymentName;
+    private final InputStream deploymentInputStream;
+    private final boolean autoCloseInputStream;
+    private final List<String> serverGroups;
+
+    private Deploy(InputStream deploymentInputStream, Data deployCmdOptions) {
```

Comment:
Agree with you, don't now about any better way with Java 6 than using the InputStream. If that would be big issue, I would redo this and allow only File. In that case only in memory handling would be lost :-(.